zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/metric/ZkClientMonitor.java

```diff
@@ -109,12 +135,8 @@ public DynamicMBeanProvider register() throws JMException {
     doRegister(attributeList, MBEAN_DESCRIPTION,
         getObjectName(_monitorType, _monitorKey, _monitorInstanceName));
     for (ZkClientPathMonitor.PredefinedPath path : ZkClientPathMonitor.PredefinedPath.values()) {
-      // If monitor root path only, check if the current path is Root.
-      // Otherwise, add monitors for every path.
-      if (!_monitorRootOnly || path.equals(ZkClientPathMonitor.PredefinedPath.Root)) {
-        _zkClientPathMonitorMap.put(path,
-            new ZkClientPathMonitor(path, _monitorType, _monitorKey, _monitorInstanceName)
-                .register());
+      if (_zkClientPathMonitorMap.get(path) != null)  {
+        _zkClientPathMonitorMap.get(path).register();
```

Comment:
nit, it could be simpler to be:

_zkClientPathMonitorMap.values().stream().foreach(...);

Reply:
There are many ways to write a loop and now we have stream. 

Still, I will just change it anyway.